src/meta/processors/job/JobManager.cpp

```diff
@@ -83,14 +83,26 @@ nebula::cpp2::ErrorCode JobManager::handleRemainingJobs() {
       auto optJob = nebula::value(optJobRet);
       std::unique_ptr<JobExecutor> je =
           JobExecutorFactory::createJobExecutor(optJob, kvStore_, adminClient_);
-      // Only balance has been recovered
+      // Only balance would change
       if (optJob.getStatus() == cpp2::JobStatus::RUNNING && je->isMetaJob()) {
-        jds.emplace_back(optJob);
+        jds.emplace_back(std::move(optJob));
+      } else if (optJob.getStatus() == cpp2::JobStatus::QUEUE) {
+        auto mutexIter = muJobFinished_.find(optJob.getSpace());
+        if (mutexIter == muJobFinished_.end()) {
+          mutexIter =
+              muJobFinished_.emplace(optJob.getSpace(), std::make_unique<std::recursive_mutex>())
+                  .first;
+        }
+        std::lock_guard<std::recursive_mutex> lk(*(mutexIter->second));
+        auto spaceId = optJob.getSpace();
+        auto jobId = optJob.getJobId();
+        enqueue(spaceId, jobId, JbOp::ADD, optJob.getJobType());
+        inFlightJobs_.emplace(std::move(jobId), std::move(optJob));
       }
     }
   }
   for (auto& jd : jds) {
-    jd.setStatus(cpp2::JobStatus::QUEUE, true);
+    jd.setStatus(cpp2::JobStatus::FAILED, true);
```

Comment:
Why you prefer FAILED instead of FAILED?

Reply:
？

Reply:
this is the rule 1